.github/workflows/workspace-integration-tests.yml

```diff
@@ -0,0 +1,179 @@
+                          echo "name=workspace-integration-test-${{ github.run_id }}-${{ github.run_attempt }}"
+                      } >> $GITHUB_OUTPUT
+                  fi
+            - name: Get Secrets from GCP
+              id: 'secrets'
+              uses: 'google-github-actions/get-secretmanager-secrets@v1'
+              with:
+                secrets: |-
+                    WORKSPACE_SLACK_WEBHOOK:gitpod-core-dev/workspace-slack-webhook
+            - name: Slack Notification
+              uses: rtCamp/action-slack-notify@v2
+              if: failure()
+              env:
+                  SLACK_WEBHOOK: ${{ steps.secrets.outputs.WORKSPACE_SLACK_WEBHOOK }}
+                  SLACK_COLOR: ${{ job.status }}
+                  SLACK_MESSAGE: main branch build failed
+    check:
+        name: Check for regressions
+        needs: [configuration]
+        runs-on: [self-hosted]
+        container:
+            image: eu.gcr.io/gitpod-core-dev/dev/dev-environment:kylos101-kubecdl-home.1
+        steps:
+            - uses: actions/checkout@v3
+            - name: Create preview environment infrastructure
+              uses: ./.github/actions/preview-create
+              with:
+                  name: ${{ needs.configuration.outputs.name }}
+                  infrastructure_provider: harvester
+                  large_vm: true
+                  sa_key: ${{ secrets.GCP_CREDENTIALS }}
+            - name: Deploy Gitpod to the preview environment
+              if: github.event.inputs.skip_deploy != 'true'
+              id: deploy-gitpod
+              uses: ./.github/actions/deploy-gitpod
+              with:
+                  name: ${{ needs.configuration.outputs.name }}
+                  sa_key: ${{ secrets.GCP_CREDENTIALS }}
+                  version: ${{ needs.configuration.outputs.version}}
+            - name: Get Secrets from GCP
+              id: 'secrets'
+              uses: 'google-github-actions/get-secretmanager-secrets@v1'
+              with:
+                secrets: |-
+                    WORKSPACE_SLACK_WEBHOOK:gitpod-core-dev/workspace-slack-webhook
+                    WORKSPACE_INTEGRATION_TEST_USERNAME:gitpod-core-dev/workspace-integration-test-username
+                    WORKSPACE_INTEGRATION_TEST_USER_TOKEN:gitpod-core-dev/workspace-integration-test-user-token
+            - name: Integration Test
+              shell: bash
+              env:
+                  ROBOQUAT_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                  USERNAME: ${{ steps.secrets.outputs.WORKSPACE_INTEGRATION_TEST_USERNAME }}
+                  USER_TOKEN: ${{ steps.secrets.outputs.WORKSPACE_INTEGRATION_TEST_USER_TOKEN }}
+                  PREVIEW_ENV_DEV_SA_KEY: ${{ secrets.GCP_CREDENTIALS }}
+                  PREVIEW_NAME: ${{ needs.configuration.outputs.name }}
+              run: |
+                  set -euo pipefail
```

Comment:
it might be cleaner to keep this script in a separate bash file, and run that file here. That way we keep e.g. syntax highlighting, bash linting, etc.

Reply:
Will do in a follow-on PR, or, if I have trouble getting having the tests run general. So, expect another PR. ;)